packages/auth/src/grant/routes.ts

```diff
@@ -190,8 +191,8 @@ async function createPendingGrant(
 
   const trx = await Grant.startTransaction()
 
+  const grant = await createGrantOrThrowError(grantService, body, trx)
```

Comment:
To simplify, we could probably:
- Only return `Promise<Grant>` in the `grantService.create` (without `GrantError`) and just let the thrown `AccessError` propagate to `grantRoutes.createPendingGrant`
- In the `createPendingGrant` catch block down below, check for `isAccessError(err)`

With this, you'll only end up adding an extra check in the  `createPendingGrant` catch block or the new error without needing to rollback the trx separately